pkg/container-collection/options.go

```diff
@@ -585,6 +582,54 @@ func WithLinuxNamespaceEnrichment() ContainerCollectionOption {
 	}
 }
 
+// WithOCIConfigEnrichment enriches container using provided OCI config
+func WithOCIConfigEnrichment() ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		cc.containerEnrichers = append(cc.containerEnrichers, func(container *Container) bool {
+			if container.OciConfig == nil || container.IsEnriched() {
+				return true
+			}
+
+			var containerTypeAnnotation, containerNameAnnotation, podNameAnnotation, podNamespaceAnnotation string
+			if cm := container.OciConfig.Annotations[crioContainerManagerAnnotation]; cm == runtimeclient.CrioName {
```

Comment:
> I wonder if it could even be a separate package, so it could be reused by [seccompagent's kuberesolver.go#L30-L41](https://github.com/kinvolk/seccompagent/blob/main/pkg/kuberesolver/kuberesolver.go#L30-L41) without importing all container-collection but that would complicate `enrichFromAnnotations()`.

True, that would be nice. How about we return a custom `struct` with all the currently supported values in a standalone package? Later on it can be mapped to `containercollection.Container` in `enrichFromAnnotations` ?

Reply:
That sounds good. Could this be done in a different PR? So that discussions on the details don't postpone this improvement on crio support.

Reply:
I opened the [PR](https://github.com/inspektor-gadget/inspektor-gadget/pull/1147) with the changes. Should we go ahead and merge this one?